### DISCUSSION TITLE
Will podVolume data restore be incremantal with --existing-resource-policy as update ?

### DISCUSSION THREAD

**Discussion by navilg** [vmware-tanzu/velero#6678] [Community support Q&A]
Hello,

We are using velero integrated with restic to backup cluster resources and data inside volume. 
We have a use case where we want to continuously restore the data from latest backup onto another cluster. Restore will include cluster resource restore as well as data inside the volume.

In velero 1.9+ --existing-resource-policy is introduced which will update the resource on cluster during restore if it already exist on cluster.

1. Does the restore will have same behavior for pod volumes as well (volume data restore) ?
2. If yes, Will this be incremental restore of volume data or will it wipe out all existing volume data and performs complete restore again ?
3. If a resource already exists (other than pod volumes, like configmap or sts), Will it replace the existing resource with the resource in backup or it will merge the definitions ?

@Lyndon-Li Could you please provide your inputs on this ?

Thanks

**Comment by Lyndon-Li** [comment]
For the files that are the same in the existing volume and the backup, they will be overwritten; for the files that are not in the existing volume but in the backup, they will be created; for the files that are in the existing volume but not in the backup, they will be left there.
If a file is to be written/overwritten, it will perform fully.

For existing resources, if the `existing-resource-policy` is set to `update`, the resources will be merged.

**Comment by Lyndon-Li** [comment]
> We have a use case where we want to continuously restore the data from latest backup onto another cluster.

If the purpose of this is to build a DR solution based on Velero's backup and restore, as you can see, there are some gaps. For a DR solution, all the changes should be synced to the DR site, i.e., add, modify, delete. 
However, as you can see, the delete operation cannot be fulfilled for both volume data and cluster resource.

**Comment by navilg** [comment]
Thanks @Lyndon-Li

**Comment by sseago** [comment]
@Lyndon-Li 
```
For the files that are the same in the existing volume and the backup, they will be overwritten; for the files that are not in the existing volume but in the backup, they will be created; for the files that are in the existing volume but not in the backup, they will be left there.
If a file is to be written/overwritten, it will perform fully.
```

Hmm. That's only true if the pod doesn't exist on restore, though. If we're not creating a new pod, then there won't be an initcontainer on the pod, so I don't think the podvolumerestore will process at all. I'm not 100% sure on this, but I think to update file contents on restore the pod (and therefore, probably the Deployment too) must be removed prior to running restore.
